src/components/footer/Footer.js

```diff
@@ -1,22 +1,67 @@
 import React from 'react';
-import { translate } from 'react-i18next';
+import { translate, Interpolate } from 'react-i18next';
 import 'i18n/i18n';
+import Panel from 'components/panel/Panel';
+import Button from 'components/button/Button';
 import styles from './footer.scss';
+import constant from 'config/constants';
 
-const Footer = ({ t }) => (
-  <footer className={styles.footerMain}>
-    <div className={styles.footerInner}>
-      <a href="#terms" className={styles.footerLink}>
-        {t('footer-terms')}
+const Footer = ({ t }) => {
+  const createTrigger = (type) => {
+    return (
+      <a href={'#' + type} className={styles.footerLink}>
+        {t(`footer-${type}`)}
       </a>
-      <a href="#cookies" className={styles.footerLink}>
-        {t('footer-cookies')}
-      </a>
-      <a href="#privacy" className={styles.footerLink}>
-        {t('footer-privacy')}
-      </a>
-    </div>
-  </footer>
-);
+    );
+  };
+
+  const createPanel = (type, detail) => (
+    <Panel trigger={createTrigger(type)} modal closable closeAction={closeBtn} key={type}>
```

Comment:
I think we don't need a `closeAction` prop, we should define a `<Footer><Button>Close</Button></Footer>` element as a child of this `<Panel>`

Reply:
Yes we can do that, but if we do that, then we need to maintain state of the panel -> open/close to toggle the panel. And for that we need to make it a container component. Right now in the footer we need to show 3 panels. If we remove this prop then we need to maintain 3 states of the panel and change the presentational footer component to presentational component.

Moreover, I feel this is really beneficial if no specific action is to be taken on closing the panel. If there is a need to perform any action then we can use `<Footer><Button onClick={abc}>Close</Button></Footer>`

Let me know your thoughts